addons/hr_holidays/models/res_partner.py

```diff
@@ -13,7 +13,7 @@ def _compute_im_status(self):
         absent_now = self._get_on_leave_ids()
         for partner in self:
             if partner.id in absent_now:
-                if partner.im_status == 'online':
+                if partner.im_status in partner.user_id._get_online_status():
```

Comment:
`user_id` on partner is not what you think it is, it's a commercial thing and it's completely unrelated to `user_ids` (the correct one)

Reply:
Nice catch, never would have found that